PWGUD/Tasks/diffMCDataScanner.cxx

```diff
@@ -0,0 +1,486 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \brief
+////  ATTENTION Nov. 2021:
+////  MFT is not implemented yet and can not be used
+////  releated code is commented
+/// \author Paul Buehler, paul.buehler@oeaw.ac.at
+/// \since  01.10.2021
+
+#include "Framework/ConfigParamSpec.h"
+
+using namespace o2;
+using namespace o2::framework;
+
+void customize(std::vector<ConfigParamSpec>& workflowOptions)
+{
+  workflowOptions.push_back(ConfigParamSpec{"runCase", VariantType::Int, 0, {"runCase: 0 - histos,  1 - mcTruth, else - tree"}});
+}
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "CommonConstants/LHCConstants.h"
+#include "Common/CCDB/TriggerAliases.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/Core/PID/PIDResponse.h"
+#include "Common/Core/MC.h"
+
+using namespace o2::framework::expressions;
+
+namespace o2::aod
+{
+namespace datascan
+{
+DECLARE_SOA_COLUMN(PID, pid, int);   //! MCTruth PID
+DECLARE_SOA_COLUMN(Pt, pt, float);   //! MCTruth particle pt
+DECLARE_SOA_COLUMN(sEL, sel, float); //! TPC nSigma el
+DECLARE_SOA_COLUMN(sMU, smu, float); //! TPC nSigma mu
+DECLARE_SOA_COLUMN(sPI, spi, float); //! TPC nSigma pi
+DECLARE_SOA_COLUMN(sKA, ska, float); //! TPC nSigma ka
+DECLARE_SOA_COLUMN(sPR, spr, float); //! TPC nSigma pr
+} // namespace datascan
+DECLARE_SOA_TABLE(nSigmas, "AOD", "NSIGMAS", //! MCTruth of particle PID and pt and TPC nSigma for el, mu, pi, ka, pr
```

Comment:
Please declare a name which contains UD both in the table name and in the table name (third argument)

Reply:
At the moment there is no task which consumes these tables. Is it planned that they are going to be consumed outside of this cxx file? In that case move the definition to PWGUD/DataModel

Reply:
ok.